Algorithms/Mathematical-Algorithms/Power/Power.cpp

```diff
@@ -0,0 +1,81 @@
+
+//https://github.com/sahilbansal17/Competitive_Coding/
+
+//Name: Anil Khatri (@imkaka)
+//Email-Id: anil.soccer.khatri@gmail.com
+
+//It calculate the power of base^power in O(log power) time complexity.
+//It is the function you need in your templete all the time.
+
+
+#include<iostream>
+#include<cstdio>
+#include<vector>
+
+using namespace std;
+//To handle large numbers for more info -> (https://codeaccepted.wordpress.com/2014/02/15/output-the-answer-modulo-109-7/)
+#define MOD 1000000007
+
+//Fast Power - Iterative versiom
+
+long long fastPower(long long base, long long power){
+
+  long long result = 1;
+
+  while(power > 0) {
+        
+        if(power & 1) { 
+            result = (result*base) % MOD;
+        }
+        base = (base * base) % MOD;
+
+        power >>= 1; 
+    }
+
+    return (result);
+
+
+}
+
+//3x3 Matrix 
+vector<vector<long long>> matrixPower(vector<vector<long long>> &matrix , int power){
+  
+  for(int i=0;i<3;i++){
+    for(int j=0;j<3;j++){
+      matrix[i][j]= fastPower(matrix[i][j], power);
+    }
+  }
+
+  return (matrix);
+}
+
+
+//main 
+int main(){
+  printf("Power: %lld\n", fastPower(2, 10000000000000));    //Outputs Power: 836339738
```

Comment:
Take inputs from user in both the programs via prompts so that we can test it.

Reply:
ok